tests/fixtures/database.py

```diff
@@ -334,7 +334,7 @@ async def deployment(
     flow_function,
     infrastructure_document_id,
     storage_document_id,
-    work_queue_1,
+    work_queue_1,  # attached to a work pool called the work_pool fixture named "test-work-pool"
```

Comment:
The reason why I created a new fixture called `deployment_in_default_work_pool` is because the currently used `deployment` fixture (that all of the agent tests were using) uses `work_queue_1` which belongs to a work pool called `test-work-pool`.

Since we are changing the agent code to only look for flow runs in the default-agent-pool if none is specified, this was causing issues as the work pool is not passed in when the agent is started during the tests. Thinking on it now, as an alternative to using the `deployment_in_default_work_pool` fixture, we could simply start the agent and pass in the desired work pool ("test-work-pool") in all of the existing tests. Any thoughts on doing one or the other, @desertaxle ?

Reply:
I lean towards passing the desired work pool to the agent for the existing tests, but we'll want to make sure that we still have a test checking for the case when a work pool is not passed in, so the `deployment_in_default_work_pool` will still be helpful.

Reply:
Great, I've added one more test that uses `deployment_in_default_work_pool` for when a work pool is not passed in 👍